src/js/components/modals/ServiceFormModal.js

```diff
@@ -0,0 +1,242 @@
+      let service = ServiceUtil.createServiceFromFormModel(model);
+      nextState.model = model;
+      nextState.service = service;
+      nextState.app = JSON.stringify(ServiceUtil
+        .getAppDefinitionFromService(service), null, 2);
+    }
+    nextState.jsonMode = !this.state.jsonMode;
+    this.setState(nextState);
+  }
+
+  onMarathonStoreServiceCreateSuccess() {
+    this.resetState();
+    this.props.onClose();
+  }
+
+  onMarathonStoreServiceCreateError(errorMessage) {
+    this.setState({
+      errorMessage
+    });
+  }
+
+  handleCancel() {
+    this.props.onClose();
+  }
+
+  handleSubmit() {
+
+    if (this.state.jsonMode) {
+      MarathonStore.createService(JSON.parse(this.state.app));
+      return;
+    }
+    if (this.triggerAdvancedSubmit) {
+      let {model} = this.triggerAdvancedSubmit();
+      let service = ServiceUtil.createServiceFromFormModel(model);
+      this.setState({
+        service: service,
+        model: model,
+        errorMessage: null
+      });
+      MarathonStore
+        .createService(ServiceUtil.getAppDefinitionFromService(service));
+    }
+  }
+
+  getAdvancedSubmit(triggerSubmit) {
+    this.triggerAdvancedSubmit = triggerSubmit;
+  }
+
+  getErrorMessage() {
+    let {errorMessage} = this.state;
+    if (!errorMessage) {
+      return null;
+    }
+
+    return (
+      <div>
+        <h4 className="text-align-center text-danger flush-top">
+          {errorMessage.message}
+        </h4>
+        <pre className="text-danger">
+          {JSON.stringify(errorMessage.details, null, 2)}
+        </pre>
+        <button
+          className="button button-small button-danger flush-bottom"
+          onClick={this.handleClearError}>
+          clear
+        </button>
+      </div>
+    );
+  }
+
+  getFooter() {
+    return (
+      <div className="button-collection flush-bottom">
+        <button
+          className="button button-large flush-top flush-bottom"
+          onClick={this.handleCancel}>
+          Cancel
+        </button>
+        <ToggleButton
+          checked={this.state.jsonMode}
+          onChange={this.handleJSONToggle}>
+          JSON mode
+        </ToggleButton>
+        <button
+          className="button button-large button-success flush-bottom"
+          onClick={this.handleSubmit}>
+          Deploy
+        </button>
+      </div>
+    );
+
+  }
+
+  getModalContents() {
+    let {app, jsonMode, service} = this.state;
+    if (jsonMode) {
+      return (
+        <Ace editorProps={{$blockScrolling: true}}
+          mode="json"
+          onChange={this.handleJSONChange}
+          showGutter={true}
+          showPrintMargin={false}
+          theme="monokai"
+          value={app}
+          width="100%"/>
+      );
+    }
+
+    return (
+      <SchemaForm
+        getTriggerSubmit={this.getAdvancedSubmit}
+        model={ServiceUtil.createFormModelFromSchema(ServiceSchema, service)}
+        onChange={Function.prototype}
```

Comment:
can you make an edit to `SchemaForm` to have a default function in this PR so you don't have to do this?